MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -861,6 +864,32 @@ public static void setScenarioMapSize(AtBDynamicScenario scenario) {
         scenario.setMapSizeX(mapSizeX);
         scenario.setMapSizeY(mapSizeY);
     }
+    
+    /**
+     * If there are maps of the appropriate size available and we roll higher than 
+     * the given threshold, replace the scenario's generated map with a fixed map from data/boards
+     */
+    private static void setScenarioMap(AtBDynamicScenario scenario, int mapChance) {
+        if ((scenario.getMapSizeX() > 0) && (scenario.getMapSizeY() > 0)) {
+            BoardClassifier bc = BoardClassifier.getInstance();
+            List<String> maps = bc.getMatchingBoards(scenario.getMapSizeX(), scenario.getMapSizeY(), 5, 5, new ArrayList<>());
+            
+            if (!maps.isEmpty() && (Compute.randomInt(100) <= mapChance)) {
+                String mapPath = Utilities.getRandomItem(maps);
+                MegaMekFile mapFile = new MegaMekFile(mapPath);
+                BoardDimensions dimensions = Board.getSize(mapFile.getFile());
+                
+                scenario.setMap(bc.getBoardPaths().get(mapPath));
+                scenario.setMapSizeX(dimensions.width());
+                scenario.setMapSizeY(dimensions.height());
+                scenario.setUsingFixedMap(true);
+                return;
+            }
+        }
```

Comment:
We should really check if we should attempt (the compute check) before determining if there are potential maps for performance reasons.